File: backend/app/schemas/research_hypothesis.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ResearchHypothesisResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    event_id: UUID
    hypothesis_text: str
    impact_chain: list[str] | None
    supporting_evidence: list[str] | None
    counter_evidence: list[str] | None
    trigger_conditions: list[str] | None
    invalidation_conditions: list[str] | None
    time_horizon: str | None = None
    risk_notes: str | None = None
    model_used: str | None = None
    status: str
    created_at: datetime

File: frontend/src/types/researchHypothesis.ts
export interface ResearchHypothesis {
  id: string;
  event_id: string;
  hypothesis_text: string;
  impact_chain: string[] | null;
  supporting_evidence: string[];
  counter_evidence: string[];
  trigger_conditions: string[];
  invalidation_conditions: string[];
  time_horizon: string;
  risk_notes: string | null;
  model_used: string;
  status: string;
  created_at: string;
}